### DISCUSSION TITLE
Add `target_position` attribute to cover entity

### DISCUSSION THREAD

**Discussion by farmio** [home-assistant/architecture#966] [Entity Models]
Many covers can report their current position while moving to a designated target position. This however can't be reflected properly in Home Assistant right now. Currently there is the current state (opening, closing, open, closed) and `current_cover_position`, but no indication where it is moving to. 

My proposal would be to add a `target_position` attribute to the cover entity.

Consumers could use this information to visualise the selected and current position individually (like a climate target temperature and current temperature). Related issue for HomeKit: https://github.com/home-assistant/core/issues/87461#issuecomment-1421954944

I did a quick research for integrations that would already have a target position value ready to be used:
- [HomeKit](https://github.com/home-assistant/core/blob/d0feb063ec3c5ec94502a7406c2811c4e47f61d9/homeassistant/components/homekit/type_covers.py#L283-L302)
- [HomeKit Device](https://github.com/home-assistant/core/blob/d0feb063ec3c5ec94502a7406c2811c4e47f61d9/homeassistant/components/homekit_controller/cover.py#L135-L136)
- [Netatmo](https://github.com/jabesq/pyatmo/blob/development/src/pyatmo/modules/module.py#L282-L283)
- [KNX](https://github.com/XKNX/xknx/blob/main/xknx/devices/cover.py#L95_L105)
- [ZwaveJS](https://github.com/home-assistant/core/blob/dev/homeassistant/components/zwave_js/cover.py#L89-L90)

I guess there are many more though as every integration that calculates the current position from travel time would probably know the target position. And if the HA service `cover.set_cover_position` was used, it would be known too.

**Comment by Jc2k** [comment]
I would love to see this, the "spring back" in the Ui when opening an Aqara cover with zigbee or homekit is annoying. I would implement it for homekit_controller in a heartbeat.

We need to have a robust plan for devices that don't have target_position data.

I don't think it's permitted to emulate data that a device doesn't natively have. I.E. I don't think based on similar proposals that guessing the target position based on the service call would be ok.

Personally I would be ok with target_position defaulting to current_position, unless the integration implements it. So status quo, unless integration implements new model.

Do we need a feature flag?
